ProteoformSuiteInternal/Proteoform.cs

```diff
@@ -52,62 +49,139 @@ public List<ExperimentalProteoform> identify_connected_experimentals()
+            TheoreticalProteoform theoretical_base, string theoretical_base_sequence, int additional_ptm_penalty)
+        {
+            List<ModificationWithMass> known_mods = theoretical_base.ExpandedProteinList.SelectMany(p => p.OneBasedPossibleLocalizedModifications.ToList()).SelectMany(kv => kv.Value).OfType<ModificationWithMass>().ToList();
+            List<PtmSet> possible_ptmsets = new List<PtmSet>();
+
+            int n_terminal_degraded_aas = degraded_aas_count(theoretical_base_sequence, ptm_set, true);
+            int c_terminal_degraded_aas = degraded_aas_count(theoretical_base_sequence, ptm_set, false);
+            foreach (PtmSet set in possible_peak_assignments)
+            {
+                List<ModificationWithMass> mods_in_set = set.ptm_combination.Select(ptm => ptm.modification).ToList();
+
+                int rank_sum = 0;
+                foreach (ModificationWithMass m in mods_in_set)
+                {
+                    if (m.monoisotopicMass == 0)
+                    {
+                        rank_sum += Lollipop.modification_ranks[m.monoisotopicMass];
+                        continue;
+                    }
+
+                    bool could_be_m_retention = m.modificationType == "AminoAcid" && m.motif.Motif == "M" && theoretical_base.begin == 2 && !mods_in_set.Contains(m);
+                    bool motif_matches_n_terminus = n_terminal_degraded_aas < theoretical_base_sequence.Length && m.motif.Motif == theoretical_base_sequence[n_terminal_degraded_aas].ToString();
+                    bool motif_matches_c_terminus = c_terminal_degraded_aas < theoretical_base_sequence.Length && m.motif.Motif == theoretical_base_sequence[theoretical_base_sequence.Length - c_terminal_degraded_aas - 1].ToString();
+                    bool cannot_be_degradation = !motif_matches_n_terminus && !motif_matches_c_terminus;
+                    if (m.modificationType == "Missing" && cannot_be_degradation
+                        || m.modificationType == "AminoAcid" && !could_be_m_retention)
+                    {
+                        rank_sum = Int32.MaxValue;
+                        break;
+                    }
+
+                    bool could_be_n_term_degradation = m.modificationType == "Missing" && motif_matches_n_terminus;
+                    bool could_be_c_term_degradation = m.modificationType == "Missing" && motif_matches_c_terminus;
+                    bool likely_cleavage_site = could_be_n_term_degradation && Lollipop.likely_cleavages.Contains(theoretical_base_sequence[n_terminal_degraded_aas].ToString())
+                        || could_be_c_term_degradation && Lollipop.likely_cleavages.Contains(theoretical_base_sequence[theoretical_base_sequence.Length - c_terminal_degraded_aas - 1].ToString());
+
+                    rank_sum -= Convert.ToInt32(Lollipop.variableModifications.Contains(m)); // favor variable modifications over regular
+
```

Comment:
I'll add a comment in the code. Briefly, these are all observed in the data, so I'd like to give them weights higher than the threshold. I/L/A cleavage is very frequently observed, so I consider those to be more likely than other cleavages. Fatty acids are special modifications that we see fairly often in the yeast data (so that's a bit hacky and hard-coded). Methionine retention isn't quite as common as those frequent cleavages, but we do observe it, so I put it down with the other cleavages.